frontend/components/buttons/Button/_styles.scss

```diff
@@ -57,6 +57,40 @@ $base-class: "button";
       $core-vibrant-blue-over,
       $core-vibrant-blue-down
     );
+    display: flex;
+  }
+  &--brand.save-loading {
+    width: 64px;
+  }
+  &--brand.update-label-loading {
+    width: 114px;
+  }
+  &--brand.save-label-loading {
+    width: 98px;
+  }
+  &--brand.save-policy-loading {
+    width: 106px;
+  }
+  &--brand.save-query-pack-loading {
+    width: 139px;
+  }
+  &--brand.add-query-loading {
+    width: 101px;
+  }
+  &--brand.save-query-loading {
+    width: 105px;
+  }
+  &--brand.create-loading {
+    width: 74px;
+  }
+  &--brand.transfer-loading {
+    width: 86px;
+  }
+  &--brand.schedule-loading {
+    width: 91px;
+  }
+  &--brand.download-installer-loading {
+    width: 154px;
```

Comment:
This approach could become pretty difficult to maintain. Is there a way we can achieve the same effect without having to set a fixed width for each button instance? 

One thought is that we don't remove the text of the button itself, we just set the opacity of the text to 0. That way, the buttons can dynamically size themselves without having to define a unique class for each instance.

Reply:
SGTM, I was going off what was already written for maintaining width, but definitely room for improvement! We could somehow set the text to transparent and overlay a spinner div on top. Or, use some ref={ref} dom code to get the width of the button and set the width using inline css.

Reply:
For sure! The initial implementation definitely has lots of room for improvement :) 

> we could somehow set the text to transparent and overlay a spinner div on top.

That sounds like the way to go. If the text is in a `span` tag, you could target the `span` and set `opacity: 0` when in the loading state. Then just put the spinner in a div on top of the span tag and center it.